packages/mui-material/src/Modal/Modal.js

```diff
@@ -112,8 +112,17 @@ const Modal = React.forwardRef(function Modal(inProps, ref) {
         ...components,
       }}
       componentsProps={{
-        root: { ...componentsProps.root, ...(!isHostComponent(Root) && { as: component, theme }) },
-        backdrop: { ...BackdropProps, ...componentsProps.backdrop },
+        root: (unstyledOwnerState) => ({
```

Comment:
Would this changes to use `useSlot`?

Reply:
Actually, after giving it some thought, I'm keen on leaving it as is. We generally don't use unstyled components to build Material or Joy components, so we won't need such an abstraction (in the case of a Modal, there already is an issue to create a hook version).